ts/tensor.go

```diff
@@ -65,14 +65,14 @@ func newTensor(ctensor lib.Ctensor, nameOpt ...string) *Tensor {
 	x.ctensor = ctensor
 	x.d = new(bigStruct)
 
+	lock.Lock()
```

Comment:
I think atomic is faster and designed for parallelism.

Reply:
Yes atomic is faster and good for concurrent memory write. (I agree it)
But in this case, that is not the point of this concurrency issue.

The bug case is:
1. (goroutine 1) count +1 -> count =1
2. (goroutine 2) count +1 -> count =2
3. (goroutine 1) name = count -> name = 2
4. (goroutine 2) name = count -> name =2 (duplicated tensor name and not freed in freeCTensor because of no name in tensor existing tensor name set)

Reply:
ah, ok. Happy to merge. Thanks